pkg/query-service/agentConf/manager.go

```diff
@@ -0,0 +1,217 @@
+			"signoz_tail_sampling": config,
+		}
+
+		opamp.AddToTracePipelineSpec("signoz_tail_sampling")
+		configHash, err := opamp.UpsertControlProcessors(ctx, "traces", processorConf, m.OnConfigUpdate)
+		if err != nil {
+			zap.S().Error("failed to call agent config update for trace processor:", err)
+			return fmt.Errorf("failed to deploy the config")
+		}
+
+		m.updateDeployStatus(ctx, ElementTypeSamplingRules, version, string(DeployInitiated), "Deployment started", configHash, configVersion.LastConf)
+	case ElementTypeDropRules:
+		var filterConfig *filterprocessor.Config
+		if err := yaml.Unmarshal([]byte(configVersion.LastConf), &filterConfig); err != nil {
+			zap.S().Error("failed to read last conf correctly", err)
+			return fmt.Errorf("failed to read the stored config correctly")
+		}
+		processorConf := map[string]interface{}{
+			"filter": filterConfig,
+		}
+
+		opamp.AddToMetricsPipelineSpec("filter")
+		configHash, err := opamp.UpsertControlProcessors(ctx, "metrics", processorConf, m.OnConfigUpdate)
+		if err != nil {
+			zap.S().Error("failed to call agent config update for trace processor:", err)
+			return err
+		}
+
+		m.updateDeployStatus(ctx, ElementTypeSamplingRules, version, string(DeployInitiated), "Deployment started", configHash, configVersion.LastConf)
+	}
+
+	return nil
+}
+
+// UpsertFilterProcessor updates the agent config with new filter processor params
+func UpsertFilterProcessor(ctx context.Context, version int, config *filterprocessor.Config) error {
+	if !atomic.CompareAndSwapUint32(&m.lock, 0, 1) {
+		return fmt.Errorf("agent updater is busy")
+	}
+	defer atomic.StoreUint32(&m.lock, 0)
+
+	// merge current config with new filter params
+	// merge current config with new filter params
+	processorConf := map[string]interface{}{
+		"filter": config,
+	}
+
+	opamp.AddToMetricsPipelineSpec("filter")
+	configHash, err := opamp.UpsertControlProcessors(ctx, "metrics", processorConf, m.OnConfigUpdate)
+	if err != nil {
+		zap.S().Error("failed to call agent config update for trace processor:", err)
+		return err
+	}
+
+	processorConfYaml, err := yaml.Marshal(config)
+	if err != nil {
+		zap.S().Warnf("unexpected error while transforming processor config to yaml", err)
+	}
+
+	m.updateDeployStatus(ctx, ElementTypeDropRules, version, string(DeployInitiated), "Deployment started", configHash, string(processorConfYaml))
+	return nil
+}
+
+// OnConfigUpdate is a callback function passed to opamp server.
+// It receives a config hash with error status.  We assume
+// successful deployment if no error is received.
+// this method is currently expected to be called only once in the lifecycle
+// but can be improved in future to accept continuous request status updates from opamp
+func (m *Manager) OnConfigUpdate(agentId string, hash string, err error) {
+
+	status := string(Deployed)
+
+	message := "deploy successful"
+
+	defer func() {
+		zap.S().Error(status, zap.String("agentId", agentId), zap.String("agentResponse", message))
```

Comment:
Should this be info instead of error as it is getting printed every time even if it's successful?

Reply:
one